### PR TITLE
Fix ios_user aggregation issue with update_password & hashed_password idempotency

### LINKED ISSUE #559: ios_user: update_password key failure with aggregate. + others
##### SUMMARY

I noticed 3 different issues with this module:

1) module breaks when the update_password is used in the aggregate dictionary.

```
  cisco.ios.ios_user:
    aggregate:
      - name: "{{ username1 }}"
        configured_password: "{{ clear passwd1 }}" 
        update_password: 'always'
      - name: "{{ username2 }}"
        configured_password: "{{ clear passwd2 }}" 
        update_password: 'always'
```
(this breaks)

works as expected is defined like this : 

```
  cisco.ios.ios_user:
    aggregate:
      - name: "{{ username1 }}"
        configured_password: "{{ clear passwd1 }}" 
      - name: "{{ username2 }}"
        configured_password: "{{ clear passwd2 }}" 
      update_password: 'always'
```
(this works)

2) some hashes are not blurred in the output with no log

expected result : 
        "username dummy secret ******** ********"
actual result:
        "username dummy secret ******** $********$human_readable_hash********",

This behaviour depends on the hash. Some are blurred others are clear.

3) ( more of an enhancement ) 
There is no choice regarding encryption.
The only algorithm available through the module is md5 which considered broken.
The only recommanded encryption algorithm, as of today, is scrypt.
Even if it is possible to configure the device with a hashed password ... it will set the same hash on all configuration files which is not desirable neither.

Proposition: add a new key word in the module to choose the encryption algorithm with clear password as input.
the command to be :
`username <username> privilege 15 algorithm-type scrypt secret <secret>`

### Work around:

Use the module "naked" with purge key set to true;
Configure desired users and passwords with config module in another task.

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
github latestansible [core 2.12.2]
  config file = /remote/users/mcarrouee/.ansible.cfg
...
  python version = 3.8.8 (default, Aug 11 2021, 06:52:42) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 3.0.3
  libyaml = True
```

##### COLLECTION VERSION

```paste below
github latest
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

module failure with aggregate

```paste below
The full traceback is:
Traceback (most recent call last):
  File "~/.ansible/tmp/ansible-local-2972609kf089_dt/ansible-tmp-1649944005.6931295-2972638-226093719612968/AnsiballZ_ios_user.py", line 107, in <module>
    _ansiballz_main()
  File "~/.ansible/tmp/ansible-local-2972609kf089_dt/ansible-tmp-1649944005.6931295-2972638-226093719612968/AnsiballZ_ios_user.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "~/.ansible/tmp/ansible-local-2972609kf089_dt/ansible-tmp-1649944005.6931295-2972638-226093719612968/AnsiballZ_ios_user.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_user', init_globals=dict(_module_fqn='ansible_collections.cisco.ios.plugins.modules.ios_user', _modlib_path=modlib_path),
  File "/usr/lib64/python3.8/runpy.py", line 207, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib64/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib64/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_cisco.ios.ios_user_payload_1zvvkg4k/ansible_cisco.ios.ios_user_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_user.py", line 621, in <module>
  File "/tmp/ansible_cisco.ios.ios_user_payload_1zvvkg4k/ansible_cisco.ios.ios_user_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_user.py", line 605, in main
  File "/tmp/ansible_cisco.ios.ios_user_payload_1zvvkg4k/ansible_cisco.ios.ios_user_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_user.py", line 550, in update_objects
KeyError: 'update_password'
```

### PR DESCRIPTION
**Pull request by TheNova22**
## Description

- What is being changed?
  
  - update_password and password_type are now resolved per aggregate item using get_param_value, allowing each entry in an aggregate list to independently override the module-level defaults.
  - A new parse_hashed_password() helper parses the existing hash type and value from running config, so repeated runs with the same hashed_password type/value pair correctly produce no commands.
  - Documentation for update_password clarifies that on_create/always applies only to configured_password; hashed_password always uses diff-based comparison.

- Why is this change needed?

  - `map_obj_to_commands` read update_password and password_type from module.params once at function entry. Per-item overrides set inside aggregate entries were silently ignored — all items shared the module-level value. If update_password was set only inside an aggregate item (not at the task level), the module-level default (always) took effect instead, causing password commands to be re-sent for existing users on every run.
  - `map_config_to_obj` always initialised hashed_password to None. Because have["hashed_password"] was always None, needs_update() treated every run as a change when want["hashed_password"] was set — re-applying the same hash type and value on every invocation.

- How does this change address the issue?

  - Per-item update_password and password_type are now stored in each want dict during map_params_to_obj via get_param_value, which handles the aggregate-level → task-level fallback automatically.
  - map_obj_to_commands reads them from want inside the per-item loop instead of from module.params before the loop.
  - parse_hashed_password() extracts {type, value} from the running config line, enabling a true diff against want["hashed_password"].

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New module (adding a new module to the collection)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Test update
- [ ] Refactoring (no functional changes)
- [ ] Collection release
- [ ] CI maintenance
- [ ] Workflow maintenance
- [ ] Configuration change

## Related Issue

- Fixes #559 

## Component Name

ios_user

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [x] I have removed all commented code (no commented code should be merged)
- [x] I have updated documentation where needed
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases
- [x] I have tested the changes in my local environment
- [x] I have verified the changes work with the target IOS version(s)
- [x] I have reviewed the acceptance criteria for related tickets

Assisted By: Claude

### CODE REVIEW

**Comment by komaldesai13** [comment]
check

File: changelogs/fragments/ios_user_aggregate_password_fixes.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    to determine whether a change is required.
+  - ios_user - documented type 9 (scrypt) as a valid hash type alongside the
+    existing type 5 (MD5) and type 8 (PBKDF2) examples for hashed_password.
+
+bugfixes:
+  - ios_user - fixed hashed_password idempotency so that re-applying the same
+    type/value pair against an already-configured user produces no commands,
+    preventing unnecessary password updates on repeat runs.
+
+minor_changes:
```
Comment (komaldesai13) [inline changelogs/fragments/ios_user_aggregate_password_fixes.yaml]: Update_password suppose to work per item in task like other sub-options under aggregate , which was not working , and hence we are fixing it , i would consider it as bugfix only

File: plugins/modules/ios_user.py [hunk 2]
```diff
@@ -64,7 +64,9 @@
           - Since passwords are encrypted in the device running config, this argument will
             instruct the module when to change the password.  When set to C(always), the
             password will always be updated in the device and when set to C(on_create) the
-            password will be updated only if the username is created.
+            password will be updated only if the username is created. This is applicable for
+            C(configured_password), while C(hashed_password) will undergo hash and type
+            comparision to check whether a change is required.
         choices:
           - on_create
```
Comment (komaldesai13) [inline plugins/modules/ios_user.py:71]: As we are mentioning this ,can we add some test for negative scenario here ? integration or unit ?

File: changelogs/fragments/ios_user_aggregate_password_fixes.yaml (hunk 1 above)
Comment (TheNova22) [reply changelogs/fragments/ios_user_aggregate_password_fixes.yaml]: Done! The change is now bugfixes + doc_changes only. Thanks!

File: plugins/modules/ios_user.py (hunk 2 above)
Comment (TheNova22) [reply plugins/modules/ios_user.py:71]: Added two new units about the statement mentioned:

- test_aggregate_hashed_password_on_create_does_not_suppress_hash_comparison
- test_aggregate_hashed_password_type_and_value_mismatch

**Review by komaldesai13 — APPROVED** [review]
